src/endpoint.cpp

```diff
@@ -1521,6 +1524,10 @@ int TcpEndpoint::write_msg(const struct buffer *pbuf)
     struct sockaddr *sock;
     socklen_t addrlen;
 
+    if (!_connected) {
+        return 0; // skip this endpoint, if not connected (e.g. during reconnect)
```

Comment:
please add comments on its own line rather than inline like this.

Reply:
Also, instead of adding a new "connected" flag, can't we just do handle the check below as a valid case? I.e.:

```
    if (fd < 0) {
        // skip this endpoint if not connected (e.g. during reconnect)
        return 0;
    }
```

Reply:
I wasn't sure, if the `fd < 0` check and the return value is needed in some other condition. But only `EPIPE` is handled specially, so you're probably right with the simplification.